pom.xml

```diff
@@ -1444,8 +1444,10 @@
       <!-- Required by Calcite -->
       <dependency>
         <groupId>org.immutables</groupId>
-        <artifactId>value-annotations</artifactId>
-        <version>${immutables.version}</version>
+        <artifactId>bom</artifactId>
```

Comment:
I think `value-annotations` should already have the annotation used in this PR. This `bom` is quite large and seems unnecessary

Reply:
We need to include both `value-annotations` and `value`. 

> This bom is quite large and seems unnecessary

I don't think it is huge, but even if it was... what is the problem? it is just a bom added to the dependency management. It just affects us if we actually use any of the dependencies listed in the bom, which are all `org.immutables` dependencies

Reply:
A bom is a lookuptable for dependencies that should always be in sync with each other. It is not downloading the dependencies yet, only when one is explicitly added as a dependency to a maven project.
The use of boms is a good practice, but the maintainer of the project must offer one.

Reply:
importing a bom is similar to manually copying the whole list of dependencies as _dependencyManangement_. That means the import itself _does not_ create a dependency to the dependencies listed in the bom. Instead, whenever we use one of the listed dependencies, the version listed in the bom will be counted in order to resolve the dependency version. As always, Maven picks the closes declaration (first the version provided in the declaration itself if it is in our pom, then the one in  dependency management, then the one in the dependency if it is a transitive one, etc).